not/localization/__init__.py

```diff
@@ -0,0 +1,98 @@
+"""The localization module provides localized user-facing strings
+
+generally, one need only import the polyglot singleton object
+
+This module uses an unorthodox approach to singletons to avoid keeping
+state in the module, which feels more awkward.
+
+When python 3.9 comes out in earnest, revisit using:
+https://www.python.org/dev/peps/pep-0603/
+"""
+
+import ctypes
+from json import load
+from locale import getdefaultlocale, normalize, windows_locale
+from os import getenv
+from pathlib import Path
+from platform import system
+from typing import Dict, Union
+
+DEFAULT_LOCALE = normalize("en")
+STRINGS_SUPPORTED = set("")
+
+
+class _Polyglot(Dict[str, str]):
+    """This class is responsible for lazy-loading the best-effort translation of user-facing
+    strings."""
+
+    language_alias_map = {
+        DEFAULT_LOCALE: {"en", "en_US"},
+    }
+    language_file_map = {
+        DEFAULT_LOCALE: "en.json",
+    }
+
+    def __init__(self) -> None:
+        system_locale = get_system_locale()
+        self.locale = next(
+            (
+                k
+                for k, v in self.language_alias_map.items()
+                if system_locale == k or system_locale in v
+            ),
+            DEFAULT_LOCALE,
+        )
+        with open(
+            Path(__file__).resolve().parent / self.language_file_map[self.locale], "r"
+        ) as file:
+            self.__dict__: Dict[str, str] = load(file)
+
+    def localized(
+        self, key: str, context: Dict[str, Union[int, Union[str, int, object]]]
+    ) -> str:
+        """Given a key string and context, return formatted localized string"""
+        return self.__dict__[key].format(**context)
+
+    def test(self) -> None:
+        """Test if the locfile is complete."""
+        for entry in STRINGS_SUPPORTED:
+            if entry not in self.__dict__:
+                raise KeyError(
+                    f"Missing required entry {entry} from {self.locale} locfile."
+                )
+
+        for entry in self.__dict__:
+            if entry not in STRINGS_SUPPORTED:
+                raise KeyError(
+                    f"Superfluous entry {entry} in {self.locale} locfile." ""
+                )
```

Comment:
absolutely fine as is, but mildly surprised you didn't reach for something like:

```python
maybe_missing_entry = (entry for entry if entry not in self.__dict__)
maybe_superfluous_entry = (entry for entry if entry not in STRINGS_SUPPORTED)

if next(maybe_missing_entry, False):
    raise KeyError(f"Missing required entry {entry} from {self.locale} locfile.")

if next(maybe_superfluous_entry, False):
    raise KeyError(f"Superfluous entry {entry} in {self.locale} locfile." "")
```